gui/tasks/forms.py

```diff
@@ -204,6 +214,31 @@ def check_rpath_exists(self):
         proc.wait()
         return proc.returncode == 0
 
+    def ssh_host_keyscan(self):
+        rkeyscan = self.cleaned_data.get("rsync_remotekeyscan")
+        if rkeyscan:
+            rhost = str(self.cleaned_data.get("rsync_remotehost"))
+            if '@' in rhost:
+                remote = rhost.split("@")
+                remote_host = remote[1]
+            else:
+                remote_host = rhost
+            rport = str(self.cleaned_data.get("rsync_remoteport"))
+            proc = subprocess.Popen([
+                "/usr/bin/ssh-keyscan",
+                "-p", str(rport),
+                "-t", "rsa",
+                str(remote_host),
+            ], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+            host_key, errmsg = proc.communicate()
+            if proc.returncode == 0 and host_key:
+                with open("/root/.ssh/known_hosts", "a") as myfile:
```

Comment:
One cannot always store this in the `root` user's known hosts file since the user initiating this task can explicitly be specified and can be different